File: dev/services/go-service/src/cache.go
package main

import (
	"context"
	"fmt"
	"github.com/go-redis/redis/v9"
	"github.com/rs/zerolog/log"
	"os"
)

var cache *redis.Client

func getCacheHostname(def string) string {

	value := os.Getenv("CACHE_HOSTNAME")
	if value == "" {
		value = def
	}
	return value
}

func getConnectionString() string {

	host := getCacheHostname("localhost")
	port := getString("cache.port", "6379")
	db := getString("cache.db", "0")
	return fmt.Sprintf("redis://:@%s:%s/%s", host, port, db)
}

func getCacheKey() string {
	return getString("cache.key", "count")
}

func connectToCache() error {
	opt, err := redis.ParseURL(getConnectionString())
	if err != nil {
		return err
	}
	cache = redis.NewClient(opt)
	return pingCache()
}

func pingCache() error {
	_, err := cache.Ping(context.Background()).Result()
	if err != nil {
		return err
	}
	log.Info().Msgf("cache connection success: %s", cache.Options().Addr)
	return nil
}

File: dev/services/python-service/src/cache.py
import redis
import config
import os
from config import get_string

def getCacheHostname(default):
  value = os.getenv("CACHE_HOSTNAME")
  if value is None:
      value = default
  return value

def connectToCache():
  host = getCacheHostname("localhost")
  port = config.get_string(["cache", "port"], 6379)
  db = config.get_string(["cache", "db"], 0)
  return redis.Redis(host=host, port=port, db=db)

cache = connectToCache()

def write():
  try:
    key = config.get_string(["cache", "key"], "count")
    cache.incrby(key, 1)
  except BaseException as err:
    print(f"Unexpected {err=}, {type(err)=}")
    raise

def read():
  try:
    key = config.get_string(["cache", "key"], "count")
    value = cache.get(key)
    if value is None:
      return 0
    return value
  except:
    raise
